snap/pack/pack_test.go

```diff
@@ -353,3 +353,14 @@ func (s *packSuite) TestPackWithCompressionUnhappy(c *C) {
 		c.Assert(snapfile, Equals, "")
 	}
 }
+
+func (s *packSuite) TestPackWithIntegrity(c *C) {
+	sourceDir := makeExampleSnapSourceDir(c, "{name: hello, version: 0}")
+
+	snapfile, err := pack.Snap(sourceDir, &pack.Options{
+		TargetDir: c.MkDir(),
+		Integrity: true,
+	})
+	c.Assert(err, IsNil)
+	c.Assert(snapfile, testutil.FilePresent)
```

Comment:
should this check that the snapfile contains some dm-verity/our own magic strings?

Reply:
Yes, added.